app/config.go

```diff
@@ -173,54 +184,76 @@ func loadExternalProperties() (map[string]interface{}, error) {
 				}
 			}
 		}
+	}
 
-		if len(props) > 0 {
-			// Resolve property values
-			resolverType := config.GetAppPropertiesValueResolver()
-			if resolverType != "" {
-				logger.Infof("'%s' is set to '%s'. ", config.ENV_APP_PROPERTY_RESOLVER_KEY, resolverType)
-				resolver := GetPropertyValueResolver(resolverType)
-				if resolver == nil {
-					errMag := fmt.Sprintf("Unsupported resolver type - %s. Resolver not registered.", resolverType)
-					return nil, errors.New(errMag)
-				}
+	resolverType := config.GetAppPropertiesValueResolver()
+	var resolver PropertyValueResolver
+	if resolverType != "" {
+		logger.Infof("'%s' is set to '%s'. ", config.ENV_APP_PROPERTY_RESOLVER_KEY, resolverType)
+		resolver = GetPropertyValueResolver(resolverType)
+		if resolver == nil {
+			errMag := fmt.Sprintf("Unsupported resolver type - %s. Resolver not registered.", resolverType)
+			return nil, errors.New(errMag)
+		}
 
-				for k, v := range  props {
-					strVal, ok := v.(string)
-					if ok && len(strVal) > 0 {
-						if strVal[0] == '$' {
-							// Use resolver
-							newVal, err := resolver.ResolveValue(strVal[1:])
-							if err != nil {
-								return nil, err
-							}
-							props[k] = newVal
-
-							// May be a secret??
-							strVal, _ = newVal.(string)
+		if len(props) > 0 {
+			// Get value using overridden property name
+			for k, v := range props {
+				strVal, ok := v.(string)
+				if ok && len(strVal) > 0 {
+					if strVal[0] == '$' {
+						// Use resolver
+						newVal, err := resolver.ResolveValue(strVal[1:])
+						if err != nil {
+							return nil, err
 						}
+						props[k] = newVal
+
+						// May be a secret??
+						strVal, _ = newVal.(string)
+					}
 
-						if len(strVal) > 0 &&  strings.HasPrefix(strVal, "SECRET:") {
-							// Resolve secret value
-							newVal, err := resolveSecretValue(strVal)
-							if err != nil {
-								return nil, err
-							}
-							props[k] = newVal
+					if len(strVal) > 0 && strings.HasPrefix(strVal, "SECRET:") {
+						// Resolve secret value
+						newVal, err := resolveSecretValue(strVal)
+						if err != nil {
+							return nil, err
 						}
+						props[k] = newVal
 					}
 				}
 			}
+		} else {
+			// Resolver is set. Get values using app prop name
+			for _, prop := range properties {
+				newVal, _ := resolver.ResolveValue(prop.Name())
+				if newVal != nil { // if resolver returns nil, default value from flogo.json will be used
+					// Use new value
+					props[prop.Name()] = newVal
+					// May be a secret??
+					strVal, _ := newVal.(string)
+					if len(strVal) > 0 && strings.HasPrefix(strVal, "SECRET:") {
+						// Resolve secret value
+						newVal, err := resolveSecretValue(strVal)
+						if err != nil {
+							return nil, err
+						}
+						props[prop.Name()] = newVal
+					}
+				} else {
+					logger.Warnf("Property '%s' could not be resolved using resolver '%s'. Using default value '%s'.", prop.Name(), resolverType, prop.Value())
```

Comment:
Since prop value may be a sensitive information like password, secret key etc, it's better not to log it.

Reply:
I removed the sensitive data from log